TestPublishIntegration flaky

Was worried about this but had never been able to trigger it:
```
15:22:08 --- FAIL: TestPublishIntegration (0.00s)
15:22:08     --- FAIL: TestPublishIntegration/Metricsets (1.17s)
15:22:08         <autogenerated>:1: 
15:22:08             	Error Trace:	integration_test.go:43
15:22:08             	            				integration_test.go:85
15:22:08             	            				integration_test.go:130
15:22:08             	Error:      	Not equal: 
15:22:08             	            	expected: "onboarding"
15:22:08             	            	actual  : "metric"
15:22:08             	            	
15:22:08             	            	Diff:
15:22:08             	            	--- Expected
15:22:08             	            	+++ Actual
15:22:08             	            	@@ -1 +1 @@
15:22:08             	            	-onboarding
15:22:08             	            	+metric
15:22:08             	Test:       	TestPublishIntegration/Metricsets
15:22:08         <autogenerated>:1: 
15:22:08             	Error Trace:	approvals.go:47
15:22:08             	            				integration_test.go:131
15:22:08             	Error:      	Test test_approved_es_documents/TestPublishIntegrationMetricsets failed with error: Received data differ from approved data. Run 'make update' and then 'approvals' to verify the Diff.
15:22:08             	Test:       	TestPublishIntegration/Metricsets
```

Somehow another event beat the onboarding document to being published during the test run.

--- a/beater/integration_test.go
+++ b/beater/integration_test.go
@@ -37,17 +37,6 @@ import (
 	"github.com/elastic/beats/libbeat/version"
 )
 
-func consumeOnboarding(t *testing.T, events <-chan beat.Event) *beat.Event {
-	select {
-	case e := <-events:
-		assert.Equal(t, "onboarding", e.Fields["processor"].(common.MapStr)["name"])
-		return &e
-	case <-time.After(time.Second):
-		assert.Fail(t, "no events received")
-	}
-	return nil
-}
-
 func collectEvents(events <-chan beat.Event, timeout time.Duration) []beat.Event {
 	var collected []beat.Event
 	for {
@@ -82,11 +71,14 @@ func testPublish(t *testing.T, apm *beater, events <-chan beat.Event, url string
 	got := body(t, rsp)
 	assert.Equal(t, http.StatusAccepted, rsp.StatusCode, got)
 
-	consumeOnboarding(t, events)
+	onboarded := false
 	var docs []map[string]interface{}
 	enc := jsonoutput.New(version.GetDefaultVersion(), jsonoutput.Config{Pretty: true})
 	for _, e := range collectEvents(events, time.Second) {
-		require.NoError(t, err)
+		if e.Fields["processor"].(common.MapStr)["name"] == "onboarding" {
+			onboarded = true
+			continue
+		}
 		adjustMissingTimestamp(&e)
 		doc, err := enc.Encode("apm-test", &e)
 		require.NoError(t, err)
@@ -97,6 +89,7 @@ func testPublish(t *testing.T, apm *beater, events <-chan beat.Event, url string
 	}
 	ret, err := json.Marshal(map[string]interface{}{"events": docs})
 	require.NoError(t, err)
+	assert.True(t, onboarded)
 	return ret
 }
 
@@ -144,7 +137,9 @@ func TestPublishIntegrationOnboarding(t *testing.T) {
 	require.NoError(t, err)
 	defer teardown()
 
-	event := consumeOnboarding(t, events)
+	allEvents := collectEvents(events, time.Second)
+	require.Equal(t, 1, len(allEvents))
+	event := allEvents[0]
 	otype, err := event.Fields.GetValue("observer.type")
 	require.NoError(t, err)
 	assert.Equal(t, "test-apm-server", otype.(string))

